src/testdrive/src/format/avro.rs

```diff
@@ -177,12 +177,21 @@ where
         .collect::<Result<Vec<_>, String>>()?;
     let mut expected = expected.iter();
     let mut actual = actual.iter();
-    for (i, (e, a)) in (&mut expected).zip(&mut actual).enumerate() {
-        if e != a {
-            return Err(format!(
-                "record {} did not match\nexpected:\n{:#?}\n\nactual:\n{:#?}",
-                i, e, a
-            ));
+    let mut index = 0..;
+    loop {
+        let i = index.next().expect("known to exist");
```

Comment:
Just for my own edification, what's the purpose of incrementing a counter like this?

Reply:
honestly none :) I was just thinking in iterators when I wrote this and that happened to seem most natural